mesonbuild/interpreter/interpreter.py

```diff
@@ -3381,3 +3383,61 @@ def func_range(self, node, args: T.Tuple[int, T.Optional[int], T.Optional[int]],
         if step < 1:
             raise InterpreterException('step must be >=1')
         return P_OBJ.RangeHolder(start, stop, step, subproject=self.subproject)
+
+    @FeatureNew('response_file', '1.2.0')
+    @typed_pos_args('response_file', varargs=(str, mesonlib.File))
+    @typed_kwargs(
+        'response_file',
+        KwargInfo('prefix', str, default='@'),
+        KwargInfo('quote', (bool, str), default=True),  # TODO: choices...
+        KwargInfo('max_args_length', (bool, int), default=127),  # TODO: sensible default?
+        KwargInfo('separator', str, default=' '),
+        KwargInfo('relative_paths', bool, default=True),
+    )
+    def func_response_file(self, node: mparser.BaseNode, args: T.Tuple[T.Union[str, mesonlib.File, T.List[mesonlib.FileOrString]]], kwargs: kwtypes.ResponseFile) -> T.Union[T.List[str], str]:
+
+        import shlex
```

Comment:
Should these imports be at the top? I'll let someone else comment on style

Reply:
Where should I move functions borrowed from ninjabackend.py? I guess interpreter should not import from there...

Reply:
Perhaps universal.py?